src/Analyzer.Utilities/JsonPathResolver.cs

```diff
@@ -71,23 +72,55 @@ public IEnumerable<IJsonPathResolver> Resolve(string jsonPath)
+        public IEnumerable<IJsonPathResolver> ResolveResourceType(string resourceType, string newCurrentPath = null, JToken newCurrentScope = null)
         {
-            string fullPath = currentPath + ".resources[*]";
+            var resourceTypeSeparator = "/";
+
+            newCurrentPath ??= this.currentPath;
+            newCurrentScope ??= this.currentScope;
+
+            string fullPath = newCurrentPath + ".resources[*]";
             if (!resolvedPaths.TryGetValue(fullPath, out var resolvedTokens))
             {
-                var resources = this.currentScope.InsensitiveTokens("resources[*]");
+                var resources = newCurrentScope.InsensitiveTokens("resources[*]");
                 resolvedTokens = resources.Select(r => (FieldContent)r).ToList();
                 resolvedPaths[fullPath] = resolvedTokens;
             }
 
+            // When trying to resolve Microsoft.Web/sites/siteextensions, for example, we should consider that siteextensions can be a child resource of Microsoft.Web/sites (a prefix of the original resource type)
+            var resourceTypePrefixes = new List<string> { resourceType };
+            var resourceTypeSuffixes = new List<string> { "" };
+            var indexesOfTypesSeparators = Regex.Matches(resourceType, resourceTypeSeparator).Cast<Match>().Select(m => m.Index).Skip(1);
+            foreach (var indexOfTypeSeparator in indexesOfTypesSeparators)
+            {
+                resourceTypePrefixes.Add(resourceType[..indexOfTypeSeparator]);
+                resourceTypeSuffixes.Add(resourceType[(indexOfTypeSeparator + 1)..]);
+            }
+            
             foreach (var resource in resolvedTokens)
             {
-                if (string.Equals(resource.Value.InsensitiveToken("type")?.Value<string>(), resourceType, StringComparison.OrdinalIgnoreCase))
+                for (int numOfPrefix = 0; numOfPrefix < resourceTypePrefixes.Count; numOfPrefix++)
                 {
-                    yield return new JsonPathResolver(resource.Value, resource.Value.Path, this.resolvedPaths);
+                    if (string.Equals(resource.Value.InsensitiveToken("type")?.Value<string>(), resourceTypePrefixes[numOfPrefix], StringComparison.OrdinalIgnoreCase))
+                    {
+                        if (String.IsNullOrEmpty(resourceTypeSuffixes[numOfPrefix]))
+                        {
+                            yield return new JsonPathResolver(resource.Value, resource.Value.Path, this.resolvedPaths);
+                        }
+                        else
+                        {
+                            // In this case we still haven't matched the suffix of the prefix found
+                            foreach(var newJsonPathResolver in ResolveResourceType(String.Concat(resourceTypePrefixes[numOfPrefix], resourceTypeSeparator, resourceTypeSuffixes[numOfPrefix]), resource.Value.Path, resource.Value))
```

Comment:
The `ResolveResourceType` public method shouldn't expose the details of the internal recursion (by having two additional parameters that are only used internally).

An easy way around this, without having to refactor the method at all, is to create a new resolver with a new scope (as was done previously) and call `ResolveResourceType` on that, implicitly using the new scope.
```cs
var subScope = new JsonPathResolver(resource.Value, resource.Value.Path, this.resolvedPaths);
var subResourceType = String.Concat(resourceTypePrefixes[numOfPrefix], resourceTypeSeparator, resourceTypeSuffixes[numOfPrefix]);
foreach(var newJsonPathResolver in subScope.ResolveResourceType(subResourceType))
{
...
}
```